### ISSUE TITLE
Javax validation: add email validation as format

### ISSUE THREAD

**Issue by hleb-albau** [victools/jsonschema-generator#8]
Add support for `@Email` to be translated into json spec email format https://json-schema.org/understanding-json-schema/reference/string.html#format 

Also, if regexp specified in annotation, it should be added as a pattern.

**Comment by CarstenWickner** [comment]
HI @hleb-albau,

supporting the "format" property out-of-the-box could be added as a separate feature in the `jsonschema-generator`.
Adding support for any particular `@Email` annotation (or one of the other supported formats) could then be added via configuration (either directly or by providing your own module implementation).

For now, you can achieve this by providing an `InstanceAttributeOverride` via:
- `SchemaGeneratorConfigBuilder.forFields().withInstanceAttributeOverride(override)` or
- `SchemaGeneratorConfigBuilder.forMethods().withInstanceAttributeOverride(override)`

Since it can already be achieved, I won't consider this a high priority but will look into it.

**Comment by CarstenWickner** [comment]
Actually, the "format" property is already supported. 😄
You can configure it via:
- `SchemaGeneratorConfigBuilder.forFields().withStringFormatResolver(formatResolver)` or
- `SchemaGeneratorConfigBuilder.forMethods().withStringFormatResolver(formatResolver)`

E.g. like this then:
```java
ObjectMapper objectMapper = new ObjectMapper();
SchemaGeneratorConfigBuilder configBuilder = new SchemaGeneratorConfigBuilder(objectMapper, OptionPreset.PLAIN_JSON);
// checking for the `@Email` annotation being present on a given field
configBuilder.forFields()
        .withStringFormatResolver(member -> member.getAnnotation(Email.class) == null ? null : "idn-email");

SchemaGeneratorConfig config = configBuilder.build();
SchemaGenerator generator = new SchemaGenerator(config);
JsonNode jsonSchema = generator.generateSchema(YourType.class);

System.out.println(jsonSchema.toString());
```

I.e. no code changes need to be done to support this here.
I'm sorry if the naming as `withStringFormatResolver()` (instead of `withFormatResolver()`) mislead you in thinking it was not yet catered for.

**Comment by CarstenWickner** [comment]
@hleb-albau, actually your request was to support the `@Email` annotation in the `javax.validation` module which is still a valid point.

What do you reckon: should it set the "email" or "idn-email" format value?

**Comment by hleb-albau** [comment]
> @hleb-albau, actually your request was to support the `@Email` annotation in the `javax.validation` module which is still a valid point.
> 
> What do you reckon: should it set the "email" or "idn-email" format value?

As for me it should be `idn-email`. Thanks!

**Comment by CarstenWickner** [comment]
HI @hleb-albau,

I'm happy to announce that this has been implemented and released under version [v3.2.0](https://github.com/victools/jsonschema-module-javax-validation/releases/tag/v3.2.0) of the `victools/jsonschema-module-javax-validation`.

The `@Email` annotation is now being interpreted as `"format": "email"` out-of-the-box.
In order to get the "ìdn-email"` value instead, you need to provide the (new) `JavaxValidationOption.PREFER_IDN_EMAIL_FORMAT` in the constructor.

The usage would look like this then:
```java
ObjectMapper objectMapper = new ObjectMapper();
JavaxValidationModule module = new JavaxValidationModule(JavaxValidationOption.PREFER_IDN_EMAIL_FORMAT);
SchemaGeneratorConfigBuilder configBuilder = new SchemaGeneratorConfigBuilder(objectMapper, OptionPreset.PLAIN_JSON)
    .with(module);
SchemaGeneratorConfig config = configBuilder.build();
SchemaGenerator generator = new SchemaGenerator(config);
JsonNode jsonSchema = generator.generateSchema(YourClass.class);

System.out.println(jsonSchema.toString());
```
